source/MaterialXGenShader/Util.cpp

```diff
@@ -697,12 +733,14 @@ void findRenderableMaterialNodes(ConstDocumentPtr doc,
 {
     for (const NodePtr& material : doc->getMaterialNodes())
     {
-        // Push the material node only once.
-        elements.push_back(material);
-
         // Scan for any upstream shader outpus and put them on the "processed" list
         // if we don't want to consider them for rendering.
         vector<NodePtr> shaderNodes = getShaderNodes(material);
+        if (!shaderNodes.empty())
+        {
+            // Push the material node only once if any shader nodes are found
+            elements.push_back(material);
+        }
         for (NodePtr shaderNode : shaderNodes)
```

Comment:
Nitpick: The for loop could be moved inside the if statement above, to not check it again if it was empty.